SpringProject/src/main/java/com/gradproject2019/userConference/controller/UserConferenceController.java

```diff
@@ -26,9 +26,15 @@
         return ResponseEntity.ok(newInterest);
     }
 
-    @GetMapping(path = "/user-conferences/{id}")
-    public ResponseEntity<List<ConferenceResponseDto>> getConferenceByUserId(@RequestHeader("Authorization") UUID token, @PathVariable("id") Long userId) {
+    @GetMapping(path = "/user-conferences/{userId}")
```

Comment:
I should've said that in the previous PR but userId here also acts as a filter, so it should be a query parameter rather than a path one, am I right Ed?

Reply:
I think that a nicer path would be `GET users/{user_id}/conferences` - this also then solves the other problem for the DELETE path - we can have `DELETE users/{user_id}/conferences/{conference_id}`. 

There is a slight problem with this suggestion - that is that we are assuming the only relationship between users and conference is "interested in", but maybe that's a problem to worry about later

Reply:
Yeah - I said this in the previous PR. The issue with the `user-conferences` is it makes it sound like an entity (when really its just an link between 2 different entities (user/conference).

I suggested having a path of `users/{id}/conferences`, but a query param of `/users` to filter conferences is equally fine. 

I think whoevers running this scheme should make sure we all understand what the difference between entities and database tables is - i.e. an entity is a business object with some intrinsic meaning, a database is a way of structing data. In this case of relational databases, e.g. MySql it structures the _relations_ between entities. Entities are the things which go in your paths, db tables are not. 

In this case a row in a db is definitely not an entity because it has no intinsic meaning (its just 2 numbers), and it just a _relation_ between users and conferences. 

I feel that forcing the writing java beans whenever you need to interact with a database has confused the shit out of everyone.

Reply:
So we just realised that we actually don't need a `{userId}` as a path variable because users should only be able to access/edit/delete/etc. their own data so we can just use the `userId` field we retrieve from the database when we check their `UUID` is valid.